packages/cli-e2e/__tests__/orgResources.specs.ts

```diff
@@ -94,25 +121,80 @@ describe('org:resources', () => {
   });
 
   describe('org:resources:preview', () => {
-    it(
-      'should preview the snapshot',
-      async () => {
-        const previewTerminal = previewChange(testOrgId, processManager);
-
-        const expectedOutput = [
-          'Extensions',
-          '\\+   1 to create',
-          'Fields',
-          '\\+   2 to create',
-          'Filters',
-          '\\+   1 to create',
-        ].join('\\s*');
-        const regex = new RegExp(expectedOutput, 'gm');
-
-        await previewTerminal.when(regex).on('stdout').do().once();
-      },
-      defaultTimeout
-    );
+    describe('when resources are synchronized', () => {
+      it(
+        'should preview the snapshot',
+        async () => {
+          const previewTerminal = previewChange(
+            testOrgId,
+            processManager,
+            'org-config-preview-sync'
+          );
+
+          const expectedOutput = [
+            'Extensions',
+            '\\+   1 to create',
+            'Fields',
+            '\\+   2 to create',
+            'Filters',
+            '\\+   1 to create',
+            'Ml models',
+            '\\+   1 to create',
+          ].join('\\s*');
+          const regex = new RegExp(expectedOutput, 'gm');
+
+          await previewTerminal.when(regex).on('stdout').do().once();
+        },
+        defaultTimeout
+      );
+    });
+
+    describe('when resources are not synchronized', () => {
+      beforeAll(async () => {
+        await createFieldWithoutUsingSnapshot(platformClient);
+      });
+
+      it(
+        'should throw a synchronization warning on a field',
+        async () => {
+          let returnedSyncronizationWarning = false;
+          let returnedPreview = false;
+
+          const previewTerminal = previewChange(
+            testOrgId,
+            processManager,
+            'org-config-preview-unsync'
+          );
+
+          const previewTerminalExitPromise = previewTerminal
+            .when('exit')
+            .on('process')
+            .do()
+            .once();
+
+          previewTerminal
+            .when(/Checking for automatic synchronization/)
+            .on('stderr')
+            .do(() => {
+              returnedSyncronizationWarning = true;
+            })
+            .until(previewTerminalExitPromise);
+
+          previewTerminal
+            .when(/Previewing resource changes/)
+            .on('stdout')
+            .do(() => {
+              returnedPreview = true;
+            })
+            .until(previewTerminalExitPromise);
+
+          await previewTerminalExitPromise;
+          expect(returnedSyncronizationWarning).toBe(true);
+          expect(returnedPreview).toBe(true);
```

Comment:
Small suggestions:
 - Typo in var name
 - Put the exitCondition at the end, simpler to understand I reckon, and no need to 'store' the promise like this.
 
```suggestion
          let returnedSynchronizationWarning = false;
          let returnedPreview = false;

          const previewTerminal = previewChange(
            testOrgId,
            processManager,
            'org-config-preview-unsync'
          );

          previewTerminal
            .when(/Checking for automatic synchronization/)
            .on('stderr')
            .do(() => {
              returnedSyncronizationWarning = true;
            })
            .until(previewTerminalExitPromise);

          previewTerminal
            .when(/Previewing resource changes/)
            .on('stdout')
            .do(() => {
              returnedPreview = true;
            })
            .until(previewTerminalExitPromise);

          await previewTerminal
            .when('exit')
            .on('process')
            .do()
            .once();

          expect(returnedSynchronizationWarning).toBe(true);
          expect(returnedPreview).toBe(true);
```

Reply:
If it is at the end, how can I use it in the `.until(` clause (lines 181 & 189 before your changes)?

Reply:
eh, you right nvm